ginga/misc/Callback.py

```diff
@@ -28,24 +34,36 @@ def has_callback(self, name):
         #return self.cb.has_key(name) and (len(self.cb[name]) > 0)
         return name in self.cb
 
+    # TODO: Should this raise CallbackError, or silently pass
     def delete_callback(self, name):
-        del self.cb[name]
-
-    def add_callback(self, name, fn, *args, **kwdargs):
         try:
-            tup = (fn, args, kwdargs)
-            if not tup in self.cb[name]:
-                self.cb[name].append(tup)
+            del self.cb[name]
         except KeyError:
             raise CallbackError("No callback category of '%s'" % (
                 name))
 
-    # TODO: to be deprecated ?
+
+    # TODO: Add a argumet validation function for a callback
+    # Pointers:
+    #      * Check the name of the event for which callback added
+    #      * Check that fn is a function
+    #      * Check fn accepts at least one argument, the calling object
+    #      * Check fn takes:
+    #           - at least len(args) + len(kwdargs) number of args
+    #           - variable number of args or keyword args
+    #      * Does and what value the callback funtion return
+    #          
+    def add_callback(self, name, fn, *args, **kwdargs):
+        self.enable_callback(name)
+        
+        tup = (fn, args, kwdargs)
+        if not tup in self.cb[name]:
+            self.cb[name].append(tup)
+        
     def set_callback(self, name, fn, *args, **kwdargs):
-        if not self.has_callback(name):
-            self.enable_callback(name)
-        return self.add_callback(name, fn, *args, **kwdargs)
+        self.add_callback(name, fn, *args, **kwdargs)
```

Comment:
`set_callback` should create the category if it doesn't exist and then add the callback.